src/render/color.rs

```diff
@@ -28,13 +48,272 @@ impl Color {
     pub fn values(&self) -> (u8, u8, u8, u8) {
         (self.red, self.green, self.blue, self.alpha)
     }
+
+    /// Parses a String into a Color
+    /// The following formats are supported:
+    /// - "RRGGBB"
+    /// - "AARRGGBB"
+    /// - "#RRGGBB"
+    /// - "#AARRGGBB"
+    /// - "0xRRGGBB"
+    /// - "0xAARRGGBB"
+    pub fn parse(s: String) -> Option<Color> {
+        if s.starts_with("#") {
+            Color::parse(s[1..].to_string())
+        } else if s.starts_with("0x") {
+            Color::parse(s[2..].to_string())
+        } else if s.len() == 8 {
+            Color::parse_argb(s)
+        } else if s.len() == 6 {
+            Color::parse_rgb(s)
+        } else {
+            None
+        }
+    }
+
+    /// Parses an ARGB String into a Color
+    fn parse_argb(s: String) -> Option<Color> {
+        if s.len() == 8 {
+            let mut chars = s.chars();
+            let alpha = Color::parse_color(&mut chars);
+            let color = Color::parse_rgb(chars.as_str().to_string());
+            if WLC_BUG_INVERT_RED_BLUE {
+		// Due to the bug, the colors are already inverted, so in the returned color
+                // red is blue and blue is red.
+                alpha.and_then(|a| color.map(|rgb| Color::rgba(rgb.blue, rgb.green, rgb.red, a)))
```

Comment:
`alpha` can be returned early so you don't need this `and_then` by simply doing above:

```rust
let alpha = Color::parse_color(&mtu chars)?;
```

Reply:
TIL 👍

Reply:
This doesn't work for the last line, does it? (eg. I'll still have to map over the last Option)

Reply:
You can do 

```rust
let color = Color::parse_rgb(chars.as_str())?
...
Some(Color::rgba(rgb.blue, rgb.green, rgb.red, a))
```